tests/generative/variable_strategies.py

```diff
@@ -85,6 +90,12 @@ class DomainConstraint:
             add: ({int, float}, DomainConstraint.ANY),
             subtract: ({int, float}, DomainConstraint.ANY),
             multiply: ({int, float}, DomainConstraint.ANY),
+            date_add_years: ({datetime.date}, DomainConstraint.ANY),
+            date_add_months: ({datetime.date}, DomainConstraint.ANY),
+            date_add_days: ({datetime.date}, DomainConstraint.ANY),
+            date_difference_in_years: ({int}, DomainConstraint.ANY),
+            date_difference_in_months: ({int}, DomainConstraint.ANY),
+            date_difference_in_days: ({int}, DomainConstraint.ANY),
         }
```

Comment:
I am not at all confident that these are in the right simplest-first order.  The year-from and to-first-of ones only take a single series input, but feel like date-related operations are probably more complex than comparisons, but negate can return more than one type which might make it more complex.  The date_add and date_diff ones take 2 series inputs and are dates so I've put them at the end, but :shrug: